client/src/components/activities/ActivityOverview/table.tsx

```diff
@@ -0,0 +1,43 @@
+type ValueAndElement<T> = { value: T; element: JSX.Element };
+
+export type Table<TData extends object, TDataKey extends keyof TData> = {
+	fields: TDataKey[];
+	rows: Record<TDataKey, ValueAndElement<TData>>[];
```

Comment:
Here, I use `TDataKey`, but below it's `TField`. Use the same one for both, they represent the same thing.